parachain/pallets/basic-channel/src/outbound/mod.rs

```diff
@@ -263,8 +288,8 @@ pub mod pallet {
 			<T as Config>::Hashing::hash(&commitment)
 		}
 
-		fn average_payload_size(messages: &[MessageOf<T>]) -> u32 {
-			let sum: usize = messages.iter().fold(0, |acc, x| acc + x.payload.len());
+		fn average_payload_size(messages: &[EnqueuedMessage<T::AccountId, T::MaxMessagePayloadSize>]) -> u32 {
```

Comment:
Just ignore or stub out `average_payload_size` for now. Its not on the critical path.